public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/situation.ts

```diff
@@ -0,0 +1,526 @@
+import type { Tree, SyntaxNode } from '@lezer/common';
```

Comment:
Here is the code to resolve from an input value to an autocompletion "situation". Each situation will require a different autocompletion set of strings.